verl/utils/kernel/flash_attn.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2025 Bytedance Ltd. and/or its affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from verl.utils.device import is_cuda_available, is_npu_available
+
+if is_cuda_available:
```

Comment:
This is not ideal because we may import these functions in pure CPU machine as well

Reply:
import `flash_attn` or `torch_npu` on cpu only machine will raise Exception, so it's ok here.

Reply:
Yes. Sometimes we import this function on CPU-only machine on driver and the object is pickled to the worker with GPU

Reply:
Maybe it's actually a better choice to copy those function out and implement in the verl repo.